buffer/src/main/java/io/netty/buffer/api/DefaultCompositeBuffer.java

```diff
@@ -360,6 +360,16 @@ public boolean readOnly() {
         return readOnly;
     }
 
+    @Override
+    public boolean isDirect() {
+        // A composite buffer is direct, if all components are direct.
+        boolean direct = true;
+        for (Buffer buf : bufs) {
+            direct &= buf.isDirect();
+        }
+        return direct;
+    }
+
```

Comment:
There is no need to continue looping if we found one `non direct` buffer:

```suggestion
    public boolean isDirect() {
        // A composite buffer is direct, if all components are direct.
        for (Buffer buf : bufs) {
            if (!buf.isDirect()) {
                return false;
            }
        }
        return true;
    }

```